esphome/components/esp32/__init__.py

```diff
@@ -172,6 +172,19 @@ def add_idf_component(
             KEY_COMPONENTS: components,
             KEY_SUBMODULES: submodules,
         }
+    else:
+        component_config = CORE.data[KEY_ESP32][KEY_COMPONENTS][name]
+        if components is not None:
+            component_config[KEY_COMPONENTS] = list(
+                set(component_config[KEY_COMPONENTS] + components)
+            )
+        if submodules is not None:
+            if component_config[KEY_SUBMODULES] is None:
+                component_config[KEY_SUBMODULES] = submodules
+            else:
+                component_config[KEY_SUBMODULES] = list(
+                    set([KEY_SUBMODULES] + submodules)
```

Comment:
Missing a critical piece :wink: 
```suggestion
                    set(component_config[KEY_SUBMODULES] + submodules)
```

Reply:
I think i was sleeping at that point :(